.github/workflows/ci.yml

```diff
@@ -0,0 +1,136 @@
+name: tests-and-publish
+
+on:
+  push:
+    # Avoid using all the resources/limits available by checking only
+    # relevant branches and tags. Other branches can be checked via PRs.
+    branches: [main]
+    tags:
+      - 'v[0-9]+\.[0-9]+\.[0-9]+\.dev[0-9]+'  # Match tags that resemble a version
+      - 'v[0-9]+\.[0-9]+\.[0-9]+'  # Match tags that resemble a version
+  pull_request:  # Run in every PR
+  workflow_dispatch:  # Allow manually triggering the workflow
+  schedule:
+    # Run roughly every 15 days at 00:00 UTC
+    # (useful to check if updates on dependencies break the package)
+    - cron: '0 0 1,16 * *'
+
+concurrency:
+  group: >-
+    ${{ github.workflow }}-${{ github.ref_type }}-
+    ${{ github.event.pull_request.number || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  prepare:
+    runs-on: ubuntu-latest
+    outputs:
+      wheel-distribution: ${{ steps.wheel-distribution.outputs.path }}
+    steps:
+      - uses: actions/checkout@v3
+        with: {fetch-depth: 0}  # deep clone for setuptools-scm
+      - uses: actions/setup-python@v4
+        id: setup-python
+        with: {python-version: "3.11"}
```

Comment:
Should we maybe drop this down to e.g. 3.8 as this project is a library which should be compatible in a lot of situations?

Reply:
Good point! I opened PR #28 to adapt our previous codebase to work with Python 3.8.